spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/AdminController.java

```diff
@@ -81,570 +81,567 @@
+		userRepository.save(user);
+
+		// Prepare response
+		ServiceWizardResponse response = new ServiceWizardResponse();
+		response.setApiKey(apiKey.getKeyValue());
+		response.setMonthlyCharge(wizard.getMonthlyCharge());
+		response.setPercallCharge(wizard.getPercallCharge());
+		response.setProxyUrl("/admin/proxies/" + proxy.getId() + "/download");
+		response.setServiceCost(wizard.getServiceCost());
+		response.setServiceForFree(wizard.getServiceForFree());
+		response.setServiceName(wizard.getServiceName());
+		response.setUriIdentifier(newServiceApi.getUriIdentifier());
+		response.setServiceUrl(wizard.getServiceUrl());
+		response.setUserName(user.getUsername());
+
+		ModelAndView mav = new ModelAndView();
+		mav.addObject("wizardResponse", response);
+		mav.setViewName("admin/wizard/serviceWizardResult");
+
+		return mav;
+	}
+
+	// This is the controller for the Consumer Getting Started Wizard
+	@RequestMapping(value = "/consumer/gettingstarted", method = GET)
+	@ResponseBody
+	public ModelAndView gettingStartedWizardForConsumer(Model model) throws IOException {
+
+		// API key will be generated later in the wizard, put null as dummy value for
+		// now
+		ApiKey newApiKey = null;
+
+		ServiceApi gettingStartedTestService = loadTestServiceForConsumerWizard();
+
+		ModelAndView mav = new ModelAndView();
+		mav.setViewName(ADMIN_CONSUMER_WIZARD);
+		mav.addObject("testService", gettingStartedTestService);
+		model.addAttribute("testApiKeys", newApiKey);
+		return mav;
+	}
+
+	// Refresh the Test Api Key and the URL to call the service in the getting
+	// started wizard for consumer
+	@RequestMapping(value = "/consumer/gettingstarted/refreshApiKeys/{whichFragmentToLoad}", method = GET)
+	public String refreshApiKeys(@PathVariable("whichFragmentToLoad") String whichFragmentToLoad, Model model) {
+
+		ServiceApi gettingStartedTestService = loadTestServiceForConsumerWizard();
+		List<ApiKey> apiKeysForTestService = apiKeyRepository
+				.findByLoggedInAPIConsumerAndServiceId(gettingStartedTestService.getId());
+
+		// Gets the last Getting Started Test Service APIKey, as the Getting Started for
+		// Consumer can be used several times (there is size -1 because the array starts
+		// in element 0)
+		ApiKey newApiKey = apiKeysForTestService.get(apiKeysForTestService.size() - 1);
+
+		// get the URL(s) for the test service proxy(s)
+		Map<String, List<String>> proxyURLPerNewApiKey = new TreeMap<>();
+		List<String> proxiesUrlList = proxyRepository
+				.customSearchForAllProxiesForGivenServiceApiId(newApiKey.getServiceApi().getId()).stream()
+				.filter(proxy -> proxy.getPublicUrl() != null).filter(proxy -> proxy.getPublicUrl() != "")
+				.map(Proxy::getPublicUrl).collect(Collectors.toList());
+
+		if (proxiesUrlList.size() > 1) {
+			log.info(
+					"Please notice the list of proxies is larger than one, so for test consumer we will take into account only first one");
+		}
+		List<String> defaultProxyList = new ArrayList();
```

Comment:
Instead of initializing a list and then add an element, there is a possibility to initialize a list with the element.
```suggestion
		List<String> defaultProxyList = Arrays.asList(proxiesUrlList.get(0));
```